clippy_lints/src/needless_pass_by_ref_mut.rs

```diff
@@ -304,6 +315,23 @@ impl<'tcx> MutablyUsedVariablesCtxt<'tcx> {
         }
         self.aliases.insert(alias, target);
     }
+
+    // The goal here is to find if the current scope is unsafe or not. It stops when it finds
+    // a function or an unsafe block.
+    fn is_in_unsafe_block(&self, mut item: HirId) -> bool {
+        let hir = self.tcx.hir();
+        while let Some(parent) = hir.get_enclosing_scope(item) {
```

Comment:
This should use `parent_iter` imo, mostly to make `item` immutable. `get_enclosing_scope` just iterates over the parents until it finds a block/item, but returns an `HirId`. Then you can call `fn_sig` on the `Node`, which is what `fn_sig_by_hir_id` does.

Reply:
I keep forgetting about `parent_iter`... Thanks for the lead!